src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -383,196 +411,318 @@ public HttpResponseMessage PrintPdf(PostedDataEntity loadDocumentRequest)
 
-            }
-            else
+            using (GroupDocs.Viewer.Viewer viewer = new GroupDocs.Viewer.Viewer(documentGuid, GetLoadOptions(password), settings))
             {
-                ImageOptions imageOptions = new ImageOptions();
-                SetOptions(imageOptions, password, page.Number);
-                byte[] bytes;
-                using (var memoryStream = new MemoryStream())
+                MemoryPageStreamFactory pageStreamFactory = new MemoryPageStreamFactory(pages);
+                if (globalConfiguration.Viewer.GetIsHtmlMode())
                 {
-                    this.GetHandler().GetPages(documentGuid, imageOptions)[0].Stream.CopyTo(memoryStream);
-                    bytes = memoryStream.ToArray();
-                }
-                string encodedImage = Convert.ToBase64String(bytes);
-                return encodedImage;
-            }
-        }
+                    ViewOptions viewOptions = HtmlViewOptions.ForEmbeddedResources(pageStreamFactory);
 
-        private List<string> GetAllPagesContent(string password, string documentGuid)
-        {
-            List<string> allPages = new List<string>();
-            if (globalConfiguration.Viewer.GetIsHtmlMode())
-            {
-                HtmlOptions htmlOptions = new HtmlOptions();
-                SetOptions(htmlOptions, password, 0);
-                // get page HTML              
-                var pages = this.GetHandler().GetPages(documentGuid, htmlOptions);
-                for (int i = 0; i < pages.Count; i++)
-                {
-                    allPages.Add(pages[i].HtmlContent);
+                    viewer.View(viewOptions, page.Number);
+
+                    return Encoding.UTF8.GetString(pages[0].ToArray());
                 }
-            }
-            else
-            {
-                ImageOptions imageOptions = new ImageOptions();
-                SetOptions(imageOptions, password, 0);
-                var pages = this.GetHandler().GetPages(documentGuid, imageOptions);
-                for (int i = 0; i < pages.Count; i++)
+                else
                 {
-                    byte[] bytes;
-                    using (var memoryStream = new MemoryStream())
-                    {
-                        pages[i].Stream.CopyTo(memoryStream);
-                        bytes = memoryStream.ToArray();
-                    }
-                    string encodedImage = Convert.ToBase64String(bytes);
-                    allPages.Add(encodedImage);
+                    PngViewOptions pngViewOptions = new PngViewOptions(pageStreamFactory);
+
+                    viewer.View(pngViewOptions, page.Number);
+
+                    return Convert.ToBase64String(pages[0].ToArray());
                 }
             }
-            return allPages;
         }
 
-        private void SetOptions(HtmlOptions options, string password, int pageNumber)
+        private void CreatePagesInfoFile(string pagesInfoPath, ViewInfo viewInfo)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'CreatePagesInfoFile' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)